cpp/src/sampling/detail/graph_functions.hpp

```diff
@@ -77,6 +77,42 @@ gather_one_hop_edgelist(
   const rmm::device_uvector<vertex_t>& active_majors,
   bool do_expensive_check = false);
 
+/**
+ * @brief Gather edge list for specified vertices
+ *
+ * Collect all the edges that are present in the adjacency lists on the current gpu
+ *
+ * @tparam GraphViewType Type of the passed non-owning graph object.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph_view Non-owning graph object.
+ * @param active_majors Device vector containing all the vertex id that are processed by
+ * gpus in the column communicator
+ * @return A tuple of device vector containing the majors, minors and weights gathered locally
```

Comment:
@return A tuple of device vectors containing the majors, minors, ids and optional weights, types, ..